features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -655,14 +718,21 @@ nsapi_error_t mbed_lwip_bringup_2(bool dhcp, bool ppp, const char *ip, const cha
        }
     }
 
+    lwip_connected = NSAPI_ERROR_IN_PROGRESS;
+
+
+    // mbed_lwip_core_init sets the NETIF_FLAG_LINK_UP, so is below semaphore useless?
```

Comment:
I think it is not useless. Where is the link up set?

Reply:
Not sure, but be aware there is some difference between PPP and Ethernet operation. When doing a PPP connect, some of the PPP code does some of the "up" flag setting you would have to do yourself for Ethernet.